lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -184,11 +187,18 @@ def self.bulk_destroy(elements)
         PolicyElementAssociation.where(object_attribute_id: id_groups[ObjectAttribute]).delete_all
       end
 
-      def self.bulk_assign(parents_and_children)
-        id_pairs = parents_and_children.map { |parent, child| [parent.id, child.id]  }
+      def self.bulk_assign(pairs_hash)
+        id_pairs = pairs_hash.values.map { |parent, child| [parent.id, child.id]  }
         Assignment.import([:parent_id, :child_id], id_pairs, on_duplicate_key_ignore: true)
       end
 
+      def self.bulk_unassign(pairs_hash)
+        pairs_str = pairs_hash.values.reduce([]) do |memo,(parent,child)|
+          parent.persisted && child.persisted? ? memo + ["(#{parent.id},#{child.id})"] : memo
+        end.join(',')
```

Comment:
The derivation of `paris_str` scares me. I'm having  a hard time picturing what it'll look like when it's done with this join. Can you provide some examples?